plugin/src/main/java/org/jenkinsci/plugins/structs/describable/DescribableModel.java

```diff
@@ -104,19 +105,39 @@
      */
     private final String[] constructorParamNames;
 
-    /** binds type parameter */
-    static <T> DescribableModel<T> of(Class<T> clazz) {
-        return new DescribableModel<T>(clazz);
+    /** Binds type parameter, preferred means of obtaining a DescribableModel. */
+    public static <T> DescribableModel<T> of(Class<T> clazz) {
+        DescribableModel mod = modelCache.get(clazz.getName());
+        if (mod != null) {
+            return mod;
+        }
+        mod = new DescribableModel<T>(clazz);
+        modelCache.put(clazz.getName(), mod);
```

Comment:
Isn't it already being added in the constructor?

Reply:
@andresrc Kind of -- it's still generating redundant instances, so this is something of a hack to work around the fact that I can't make the constructor private.